### DISCUSSION TITLE
WebGL fingerprinting profile setting

### DISCUSSION THREAD

**Discussion by nradosevic** [kaliiiiiiiiii/Selenium-Profiles#33] [Q&A] [answered]
I just realized that navigator.hardwareConcurrency is different depending on the device, and it is well known for mobile devices, for example it's 6 for all iPhones and all iPads older then 2018, and it's 8 newer ones and for most Android devices

And that's also the case for

WebGLRenderingContext.getSupportedExtensions()
WebGLRenderingContext.getParameter()

as WebGL Vendor and WebGL Renderer are different across different devices and again known for mobile devices, and can be used to detect if data is inconsistent

**Comment by ttlns** [comment, marked as answer]
Yeah, I am aware of that.
- `navigator.hardwareConcurrency` I can implement easely using the [chrome-developer-protocoll](https://chromedevtools.github.io/devtools-protocol/tot/Emulation/#method-setHardwareConcurrencyOverride)
- WebGl fingerprinting is more challenging. Because currently, I don't have any way to overwrite javascript objects on chrome, without any possibility of getting detected. [resource](https://adtechmadness.wordpress.com/2019/03/23/javascript-tampering-detection-and-stealth/)

**Reply by ttlns** [reply]
@nradosevic `navigator.hardwareConcurrency` implemented with new version (https://github.com/kaliiiiiiiiii/Selenium-Profiles/commit/9f479048aa4099f30c60331b0e5dfcb25bda924d) 

Note: Seems not to be implemented on chrome-90.0... (=> google-colab)

**Comment by ttlns** [comment]
Closing as it seems to be resolved
